src/Couchbase.Lite.Shared/Query/N1QLQuery.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Couchbase.Lite.Query;
+using Couchbase.Lite.Support;
+using JetBrains.Annotations;
+using LiteCore.Interop;
+using System;
+using System.Collections.Generic;
+
+namespace Couchbase.Lite.Internal.Query
+{
+    internal class N1QLQuery : XQuery
```

Comment:
Not really wild about this inheritance scheme.  A N1QL query is not really a specialization of XQuery, but a sibling to it.  I think this could be refactored better, but I will just leave this as a comment.

Reply:
Created a QueryBase class to share between XQuery and NQuery.